website/content/docs/connect/gateways/mesh-gateway/wan-federation-via-mesh-gateways.mdx

```diff
@@ -184,3 +184,17 @@ expected result:
 - Ensure any API request that activates datacenter request forwarding. such as
   [`/v1/catalog/services?dc=<OTHER_DATACENTER_NAME>`](/api-docs/catalog#dc-1)
   succeeds.
+
+### Upgrading the primary gateways
+
+Once federation has been established, updates for the addresses of new primary
+gateways are propagated through the gateways in the primary themselves. If the primary
```

Comment:
No it's not, the updates that need to be propagated are address changes for mesh gateways in the primary datacenter. But your question helped me think of a different way to clarify this.

I pushed an update to make explicit that this propagation is actually initiated by the secondary datacenters. 

Here's an example that helped me think through this:
1. Primary datacenter has three mesh gateways: [1,2,3]
2. Secondary datacenter is continuously dialing the primary datacenter via gateways [1,2,3] to receive updated primary gateway addresses.
3. Primary datacenter quickly swaps gateways [1,2,3] for [4,5,6], taking down [1,2,3].

If step 3 happens fast enough, then the queries from the secondary datacenter in step 2 do not return addresses of [4,5,6] before [1,2,3] become unavailable. Step 2 will continue to dial [1,2,3] because those are the only addresses it knows about, but because they're gone that means the primary datacenter is unreachable.